superset/sql_parse.py

```diff
@@ -203,13 +211,32 @@ def __init__(self, sql_statement: str, strip_comments: bool = False):
 
     @property
     def tables(self) -> Set[Table]:
+        def deepest_alias_level(alias_name: str) -> int:
+            alias_entries = [a for a in alias_name_tuples if str(a.name) == alias_name]
+            if len(alias_entries) > 0:
+                return max(alias_entries, key=lambda a: a.level).level
+            return -1
+
         if not self._tables:
             for statement in self._parsed:
-                self._extract_from_token(statement)
-
-            self._tables = {
-                table for table in self._tables if str(table) not in self._alias_names
-            }
+                result = self._extract_from_token(statement)
+                if result:
+                    (table_tuples, alias_name_tuples) = result
```

Comment:
Continuing my naming rant.. 😄 Here we coud probably just call then `tables` and `aliases`, and then later extract the `table_names`/`table_levels` from them etc.